qcengine/programs/openmm.py

```diff
@@ -0,0 +1,182 @@
+        
+        ret_data["success"] = False
+
+        # Set workdir to scratch
+        # Location resolution order config.scratch_dir, /tmp
+        parent = config.scratch_directory
+        with temporary_directory(parent=parent, suffix="_openmm_scratch") as tmpdir:
+
+            # Grab molecule, forcefield
+            jmol = input_data.molecule
+            
+            # TODO: If urls are supported by
+            # `openforcefield.typing.engines.smirnoff.ForceField` already, we
+            # can eliminate the `offxml` and `url` distinction
+            # URL processing can happen there
+            if input_data.model.offxml:
+                offxml = input_data.offxml
+            elif input_data.model.url:
+                # TODO: add pull of offxml data from a url
+                pass
+            else:
+                raise InputError("OpenMM requires either `model.offxml` or `model.url` to be set")
+
+            # Process molecule with RDKit
+            rdkit_mol = self._process_molecule_rdkit(jmol)
+
+            # Create an OFF Molecule from the RDKit Molecule
+            off_mol = Molecule(rdkit_mol)
+
+            # Load an OFF Force Field
+            off_forcefield = ForceField(offxml)
+
+            # Create OpenMM system in vacuum from forcefield, molecule
+            off_top = off_mol.to_topology()
+            openmm_top = off_top.to_openmm()
+            openmm_system = off_forcefield.create_openmm_system(off_top)
+            
+            # Need an integrator for simulation even if we don't end up using it really
+            integrator = LangevinIntegrator(300*units.kelvin, 1/units.picosecond, 0.002*units.picoseconds)
+
+            # Set platform to CPU explicitly
+            platform = openmm.Platform.getPlatformByName('CPU')
+
+            # Initialize simulation
+            simulation = Simulation(openmm_top, openmm_system, integrator, platform)
+
+            # TODO: need a way to generate positions from the topology
+            # likely an existing tool in the modules we're using since this is
+            # just a molecule floating in space with no solvent
+            simulation.context.setPositions(pdb.positions)
+
+            # Execute driver
+            try:
+                if input_data.driver == "energy":
+                    simulation.context.setPositions(pdb.positions)
+                    state = simulation.context.getState(getEnergy=True)
+                    ret_data["return_result"] = state.getPotentialEnergy()
+                elif input_data.driver == "gradient":
+                    # TODO: calculate forces with OpenMM
+                    pass
+                else:
+                    raise InputError(f"OpenMM can only compute energy and gradient driver methods. Found {input_data.driver}.")
+            except:
+                raise
+            else:
+                success = True
+
+            # GET OUTPUT, PUT INTO FORM WE NEED
```

Comment:
What output is necessary here?

Reply:
At this point, I don't believe we need anything further. I've cut this bit out (was lifted originally from the `psi4` harness implementation)